### DISCUSSION TITLE
Building a Blog with Next.js

### DISCUSSION THREAD

**Discussion by TimoStaudinger** [vercel/next.js#10691] [Help] [answered]
Hey folks!
I'm looking to use Next to build a blog.

Here's what I have in mind:

- Some folder that I can add md/mdx files to, with front matter to define author, timestamp etc.
- A homepage that automatically picks up the 3 to 5 most recent posts and shows an abstract
- A posts page that shows, well, all posts
- A post page that shows a full post with header etc.
- I should not have to update the homepage manually when I publish a post. That should happen dynamically.

Now, there's several starter kits out there that do just that, but most of them seem either pretty over-engineered, not particularly nice to use, or using an archaic version of Next.

I was wondering if there's some a recent, authoritative example out there for a use case like this?

The Next docs start building a blog, but unfortunately don't go far enough to get to the point where it's fully dynamic, unless I missed some part. Is that by design, and Next is completely unopinionated in that regard? Are there any "best practices"?

Thanks a ton!

**Comment by foxundermoon** [comment]
GitHub action should be help you

https://github.com/WeFoxTech/website/blob/master/.github/workflows/auto_release.yml

**Comment by Timer** [comment, marked as answer]
You can refer to my personal blog as an example:
https://github.com/timer/blog

New posts are created by creating a new markdown file!

**Reply by TimoStaudinger** [reply]
Thanks!

I'm assuming that `unstable_getStaticProps` is executed at build time, not at runtime, and the posts are compiled into a JSON format or so at build?

**Reply by Timer** [reply]
Correct!

**Reply by Timer** [reply]
Example file:
https://timer.blog/_next/data/0l9mFPgBEv8D37qlL7kfk/post/practical-xargs.json

**Reply by TimoStaudinger** [reply]
Nice! Judging by the `unstable_` prefix, that's not (yet) documented, right?

Really appreciate your input!!

**Reply by Timer** [reply]
We're expecting to release (and document) the late this week or early next week. You can read more about it here: https://github.com/zeit/next.js/issues/9524.

**Reply by TimoStaudinger** [reply]
Neat, I like it! :)
